src/PowerShellEditorServices/Language/FoldingReference.cs

```diff
@@ -60,4 +61,39 @@ public class FoldingReference: IComparable<FoldingReference>
             return string.Compare(this.Kind, that.Kind);
         }
     }
+
+    /// <summary>
+    /// A class that holds a list of FoldingReferences and ensures that when adding a reference that the
+    /// folding rules are obeyed, e.g. Only one fold per start line
+    /// </summary>
+    public class FoldingReferenceList : Dictionary<int, FoldingReference>
```

Comment:
If this doesn't require the all the dictionary methods, I would opt for composition over inheritance, so that no methods are exposed that would allow an invalid state in the list.